sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/metadata/StateMetadataSource.scala

```diff
@@ -104,7 +104,15 @@ class StateMetadataTable extends Table with SupportsRead with SupportsMetadataCo
       if (!options.containsKey("path")) {
         throw StateDataSourceErrors.requiredOptionUnspecified(PATH)
       }
-      new StateMetadataScan(options.get("path"))
+
+      val checkpointLocation = options.get("path")
+      var batchId = Option(options.get("batchId")).map(_.toLong)
```

Comment:
Have we documented those options?

Reply:
@HyukjinKwon - i thought I needed to do docs changes in a separate/follow-up PR. Is it ok to update as part of this PR itself ?

Reply:
I'm OK either way, but if you want to address this as follow-up PR, let's explicitly mention about the plan.

Reply:
I see you'll need to introduce more options - shall we file a JIRA ticket to document options we are going to add to support transformWithState?

Reply:
Done - created a ticket for this and linked it